core/src/main/java/io/github/tcdl/collector/Collector.java

```diff
@@ -49,23 +46,25 @@
     private TimeoutManager timeoutManager;
 
     private Clock clock;
-
     private Message requestMessage;
 
+
     private Optional<Callback<Payload>> onResponse = Optional.empty();
     private Optional<Callback<Acknowledge>> onAcknowledge = Optional.empty();
     private Optional<Callback<List<Message>>> onEnd = Optional.empty();
 
     private  ScheduledFuture ackTimeoutFuture;
     private  ScheduledFuture responseTimeoutFuture;
-    private CollectorManager collectorManager;
+    private  CollectorManager collectorManager;
+
+    public Collector(String topic, Message requestMessage, RequestOptions requestOptions, MsbContextImpl msbContext, EventHandlers eventHandlers) {
+        this.requestMessage = requestMessage;
 
-    public Collector(RequestOptions requestOptions, MsbContextImpl msbContext, EventHandlers eventHandlers) {
-        this.channelManager = msbContext.getChannelManager();
         this.clock = msbContext.getClock();
+        this.collectorManager = msbContext.getCollectorManagerFactory().findOrCreateCollectorManager(topic);
```

Comment:
Couldn't manager be injected here? This way Collector would not create its CollectorManager